### DISCUSSION TITLE
Vue exampe

### DISCUSSION THREAD

**Discussion by kmg100** [k-yle/rtsp-relay#157] [Q&A] [answered]
Hello, I am having problems implementing the module in my VUE code. Everytime i run the loadPlayer function i do not recieve anything back.

 <template>
<div>
    <canvas ref={video}></canvas>
</div>
</template>
<script>

import { loadPlayer } from 'rtsp-relay/browser';
import { ref, watchEffect } from "vue";

export default {
  props: {
    text: String,
  },

setup() {
  const video = ref(null);
  watchEffect(() => {
    if (!video.value) throw new Error('Ref is null');
    loadPlayer({
      url: 'ws://localhost:8554/mystream',
      canvas: video.value,
    });
  });
  return{
    video
  }
  }
}
</script>
<style scoped>
</style>

**Comment by k-yle** [comment, marked as answer]
This works well for me: 
```vue
<template>
  <canvas ref="canvas" />
</template>

<script setup>
import { ref, onMounted } from "vue";
import { loadPlayer } from "rtsp-relay/browser";

const canvas = ref(null);

onMounted(() => {
  loadPlayer({
    url: "ws://localhost:2000/api/stream/2",
    canvas: canvas.value,
  });
});
</script>
```

**Comment by harithirfaan** [comment]
Will it still work if I use a rtsp url instead of ws?

**Reply by k-yle** [reply]
no, the web client needs to connect to the relay server using websockets

```mermaid
flowchart LR
    Website -->|ws://| RelayServer[Relay Sever];
    RelayServer -->|rtsp://| VideoSource[Video Source]
```

**Reply by harithirfaan** [reply]
I am working with Vue framework and I am trying to embed an RTSP stream from my IP camera to my webpage. My RTSP url is rtsp://admin:Admin1234@192.168.1.136:554/Streaming/Channels/101/ and its able to stream on VLC player.

Can I clarify whether I need the code snippet from [#Example](https://github.com/k-yle/rtsp-relay#example)?
How do I obtain the websocket link in your Vue example?

Sorry I am new to web development.

**Reply by k-yle** [reply]
> Can I clarify whether I need the code snippet from [#Example](https://github.com/k-yle/rtsp-relay#example)?

The code in [#Example](https://github.com/k-yle/rtsp-relay#example) runs on the server. That code is the "Relay Server". Yes, you do need it. 

> How do I obtain the websocket link in your Vue example?

That depends on how and where you host your webserver. In the example code, the server is hosted on http://localhost:2000

**Reply by harithirfaan** [reply]
Do you have an example of the Vue code written in Options API?

**Reply by k-yle** [reply]
something like this maybe? [source](https://vuejs.org/guide/essentials/template-refs.html#accessing-the-refs)

```vue
<template>
  <canvas ref="canvas" />
</template>

<script>
import { loadPlayer } from 'rtsp-relay/browser';

export default {
  mounted() {
    loadPlayer({
      url: 'ws://localhost:2000/api/stream/2',
      canvas: this.$refs.canvas,
    });
  }
}
</script>
```
